plugin/path_config.go

```diff
@@ -64,36 +102,28 @@ func (b *GcpAuthBackend) pathConfigWrite(ctx context.Context, req *logical.Reque
 	}
 
 	c, err := b.config(ctx, req.Storage)
-
 	if err != nil {
 		return nil, err
 	}
-	if c == nil {
-		c = &gcpConfig{}
-	}
 
-	changed, err := c.Update(d)
-	if err != nil {
+	if err := c.Update(d); err != nil {
```

Comment:
I stripped the `changed` field being returned because when auth metadata fields from the user are parsed, the SDK doesn't return whether it's changed. So, there was a choice of either adding that to the SDK, or stripping the field and always updating the config. I went with stripping the field because 1) no other plugin I've seen only stores things if a field changed, and 2) if people are creating or updating a config, it's probably safe to assume it's intended to change.